server/src/main/java/org/opensearch/index/compositeindex/MetricType.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * Supported metric types for composite index
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public enum MetricType {
+    COUNT("count"),
+    AVG("avg"),
+    SUM("sum"),
+    MIN("min"),
+    MAX("max");
```

Comment:
Enums are serialized and deserialized as ordinals iirc. It may be a good idea to model these as string constants to keep them safe from any changes in order, etc